root/programs/MultiPurposeAuthSite/MultiPurposeAuthSite/Controllers/AccountController.cs

```diff
@@ -0,0 +1,2237 @@
+                {
+                    //　約款なし
+
+                    // アクティベーション
+                    IdentityResult result = await UserManager.ConfirmEmailAsync(userId, code);
+
+                    // メアド検証結果 ( "EmailConfirmation" or "Error"
+                    if (result.Succeeded)
+                    {
+                        // オペレーション・トレース・ログ出力
+                        ApplicationUser user = await UserManager.FindByIdAsync(userId);
+                        Logging.MyOperationTrace(string.Format("{0}({1}) has confirmed.", user.Id, user.UserName));
+
+                        return View("EmailConfirmation");
+                    }
+                    else
+                    {
+                        // エラー画面
+                        return View("Error");
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// メアド検証画面（約款）
+        /// POST: /Account/EmailConfirmation
+        /// </summary>
+        /// <param name="userId">string</param>
+        /// <param name="code">string</param>
+        /// <returns>ActionResultを非同期に返す</returns>
+        [HttpPost]
+        [AllowAnonymous]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> EmailConfirmation(AccountAgreementViewModel model)
+        {
+            if (ASPNETIdentityConfig.DisplayAgreementScreen)
+            {
+                // AccountAgreementViewModelの検証
+                if (ModelState.IsValid)
+                {
+                    // AccountAgreementViewModelの検証に成功
+
+                    if (ASPNETIdentityConfig.RequireUniqueEmail)
+                    {
+                        if (model.AcceptedAgreement)
+                        {
+                            // 同意された。
+
+                            // アクティベーション
+                            IdentityResult result = await UserManager.ConfirmEmailAsync(model.UserId, model.Code);
+
+                            // メアド検証結果 ( "EmailConfirmation" or "Error"
+                            if (result.Succeeded)
+                            {
+                                ApplicationUser user = await UserManager.FindByIdAsync(model.UserId);
+
+                                // メールの送信
+                                this.SendRegisterCompletedEmail(user);
+
+                                // オペレーション・トレース・ログ出力
+                                Logging.MyOperationTrace(string.Format("{0}({1}) has been activated.", user.Id, user.UserName));
+
+                                // 完了画面
+                                return View("EmailConfirmation");
+                            }
+                            else
+                            {
+                                // 失敗
+                                this.AddErrors(result);
+                            }
+                        }
+                        else
+                        {
+                            // 同意されていない。
+                            // 必要であればエラーメッセージを設定。
```

Comment:
@daisukenishino 

To find the customize point easily, it might be a good idea to revise as follows:

```cs
//todo: 必要に応じて、エラーメッセージの表示を検討してください。
```